web-ng/src/app/components/map/map.component.ts

```diff
@@ -217,7 +215,7 @@ export class MapComponent implements AfterViewInit, OnDestroy {
       this.selectMarker(marker);
       this.navigationService.setFeatureId(feature.id);
       this.centerViewportOnMarker(options);
-      this.setMapZoomDefault();
+      this.setMapZoom(zoomedInLevel);
```

Comment:
Nit: There's not hint in this line that zoomedInLevel is a constant, and that setMapZoom will only have an effect if zoomed out (the naming "setMapZoom" makes it sound like a plain-old mutator method).

Also, "centerViewportOnMarker" should only be provided with what it needs to work (min. surface area) rather than the whole `options` object.

To make the code more self-describing I'd suggest merging centerViewportOnMarker and setMapZoom into a single class method called "panAndZoom(lat, lng)". That method can access the `zoomedInLevel` constant directly. In future PRs we should probably rename static constant to all UPPER CASE.

Reply:
Hi @DakshSaini! Would you be able to incorporate this small refactor into this PR? If not please let me know and we can do it as a fast-follow after merge. Thanks

Reply:
Hi @gino-m, my sincere apologies for the delay. I have made the required changes now.

Reply:
No worries at all, happy to have your contribution!